### PR TITLE
kconfig: tfm: support for custom CMake flags when building TF-M

### PR DESCRIPTION
**Pull request by tejlmand**
This commit allows a user or subsystem to specify additional CMake flags
to be given to the TF-M build.

The CMake flags can be provided in two ways:
- TFM_EXTRA_CMAKE_FLAGS_KCONFIG: which allows for additional CMake flags
  as a space separeted list, like: "-DARG0=val0 -DARG1=val1"
- TFM_EXTRA_CMAKE_FLAGS_CMAKE: which allows a CMake subsystem to specify
  additional flags on the `tfm_build_target` using the property
  `CMAKE_FLAGS`.

The Kconfig option allows users to set flags, whereas the CMake options
allows for more complex build integration, such as the usage of
generator expressions.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

### CODE REVIEW

File: modules/Kconfig.tfm [hunk 1]
```diff
@@ -162,4 +162,36 @@ config ROM_START_OFFSET
...
+
+endchoice
+
+config TFM_EXTRA_CMAKE_FLAGS_KCONFIG_STRING
+	string "TF-M Extra CMake flags"
+	depends on TFM_EXTRA_CMAKE_FLAGS_KCONFIG
+	help
+	  Additional CMake flags to be provided when building TF-M.
+	  Those are given as regular CMake arguments with space as space
+	  seperator, like "-DARG0=val0 -DARG1=val1".
```
Comment (frkv) [inline modules/Kconfig.tfm:195]: typo: seperator => separator

File: modules/Kconfig.tfm [hunk 2]
```diff
@@ -162,4 +162,36 @@ config ROM_START_OFFSET
...
+	help
+	  Additional CMake flags are provided through the Kconfig option
+	  TFM_EXTRA_CMAKE_FLAGS_KCONFIG_STRING.
+
+config TFM_EXTRA_CMAKE_FLAGS_CMAKE
+	bool "CMake"
+	help
+	  Additional CMake flags are provided through setting the CMAKE_FLAGS
+	  property on the CMake target `tfm_build_target`.
+	  The TF-M will will be configured with a build dependency to
```
Comment (frkv) [inline modules/Kconfig.tfm:183]: "The TF-M" => TF-M

File: modules/Kconfig.tfm [hunk 3]
```diff
@@ -162,4 +162,36 @@ config ROM_START_OFFSET
...
+	  `tfm_build_target` and additional dependencies can thus be constructed
```
Comment (frkv) [inline modules/Kconfig.tfm:184]: `tfm_build_target`. Additional dependencies can be constructed

File: modules/Kconfig.tfm [hunk 4]
```diff
@@ -162,4 +162,36 @@ config ROM_START_OFFSET
 
 endif # !TFM_BL2_FALSE
 
+choice TFM_EXTRA_CMAKE_FLAGS
+	prompt "TF-M Extra CMake build flags"
+	help
+	  This choice selects selects how additional CMake flags can be passed
+	  to the TF-M CMake configure stage and which are not covered
```
Comment (frkv) [inline modules/Kconfig.tfm:169]: "and which are not covered" => ". Additional CMake flags are not covered"?

File: modules/Kconfig.tfm [hunk 5]
```diff
@@ -162,4 +162,36 @@ config ROM_START_OFFSET
...
+	  using `add_dependencies(tfm_build_target <dependency>)`.
+
+endchoice
+
+config TFM_EXTRA_CMAKE_FLAGS_KCONFIG_STRING
+	string "TF-M Extra CMake flags"
+	depends on TFM_EXTRA_CMAKE_FLAGS_KCONFIG
+	help
+	  Additional CMake flags to be provided when building TF-M.
+	  Those are given as regular CMake arguments with space as space
```
Comment (frkv) [inline modules/Kconfig.tfm:194]: "with space as a separator" instead?

File: modules/Kconfig.tfm (hunk 2 above)
Comment (tejlmand) [reply modules/Kconfig.tfm]: fixed

**Pull request closed without merging**